R/pool.R

```diff
@@ -45,118 +53,178 @@ Pool <- R6Class("Pool",
       }
       reg.finalizer(self,
         function(self) {
-          protectDefaultScheduler({
-            freeEnv <- private$freeObjects
-            if (length(freeEnv) > 0) {
-              self$close()
-            }
-          })
+          if (self$valid) {
+            warning("Closing leaked pool.")
```

Comment:
update for future reference: we're keeping things as is for now. Pooled objects must be tolerant of multiple closings since they may have multiple finalizers. If this becomes as issue in the future, we may simply add an argument to pool's `initialize` that specifies whether or not `pool` should implement finalizers (whose default would `TRUE`, pool should implement finalizers).